src/request.ts

```diff
@@ -1,207 +1,47 @@
-import Promise from 'dojo-shim/Promise';
 import Task from './async/Task';
-import has from './request/has';
-import { Handle } from 'dojo-interfaces/core';
-import MatchRegistry, { Test } from './MatchRegistry';
-import load from './load';
-import { ParamList } from './UrlSearchParams';
+import { RequestOptions, Response, Provider } from './request/interfaces';
+import ProviderRegistry from './request/ProviderRegistry';
+import xhr from './request/providers/xhr';
 
-declare var require: any;
-
-export class FilterRegistry extends MatchRegistry<RequestFilter> {
-	register(test: string | RegExp | RequestFilterTest | null, value: RequestFilter, first?: boolean): Handle {
-		let entryTest: Test;
-		const inTest = test;
-
-		if (typeof inTest === 'string') {
-			entryTest = (response, url, options) => {
-				return inTest === url;
-			};
-		}
-		else if (inTest instanceof RegExp) {
-			entryTest = (response, url, options) => {
-				return inTest.test(url);
-			};
-		}
-		else {
-			entryTest = <RequestFilterTest> inTest;
-		}
+export const providerRegistry = new ProviderRegistry();
 
-		return super.register(entryTest, value, first);
+const request: {
```

Comment:
Any reason why we don't also support `OPTIONS` and `HEAD`? At the moment we support a list that is neither the CORS safelist of verbs or the verbs subject to normalization in the spec.